File: templates/androidstarters-java/app/src/main/java/io/mvpstarter/sample/injection/component/FragmentComponent.java
package <%= appPackage %>.injection.component;

import dagger.Subcomponent;
import <%= appPackage %>.injection.PerFragment;
import <%= appPackage %>.injection.module.FragmentModule;

/**
 * This component inject dependencies to all Fragments across the application
 */
@PerFragment
@Subcomponent(modules = FragmentModule.class)
public interface FragmentComponent {
}

File: templates/androidstarters-kotlin/app/src/main/java/io/mvpstarter/sample/injection/component/FragmentComponent.kt
package <%= appPackage %>.injection.component

import <%= appPackage %>.injection.PerFragment
import <%= appPackage %>.injection.module.FragmentModule
import dagger.Subcomponent

/**
 * This component inject dependencies to all Fragments across the application
 */
@PerFragment
@Subcomponent(modules = arrayOf(FragmentModule::class))
interface FragmentComponent